lib/os/fdtable.c

```diff
@@ -63,7 +63,19 @@ static int z_fd_ref(int fd)
 
 static int z_fd_unref(int fd)
 {
-	int old_rc = atomic_dec(&fdtable[fd].refcount);
+	atomic_val_t old_rc;
+
+	/* Reference counter must be checked to avoid decrement refcount below
+	 * zero causing file descriptor leak. Loop statement below executes
+	 * atomic decrement if refcount value is grater than zero. Otherwise,
+	 * refcount is not going to be written.
+	 */
+	do {
+		old_rc = atomic_get(&fdtable[fd].refcount);
+		if (!old_rc) {
```

Comment:
If the ref count is already 0, then we should not touch anything because the fd entry is already "freed". So we should probably just do `return 0;` here.
So if we do break here, then we would return -1 in few lines below, which does not sound right.

Reply:
Right. What do you think about changing return value to void. Actually return value is not used anywhere...

Reply:
The return value can be used in debugging, but it probably has no other purpose.

Reply:
Ok. Lets keep this MR simple and fixing only incorrect behavior. 

Do you think it is worth to create another MR witch `z_fd_ref/z_fd_unref` return type change to void?

Reply:
> Do you think it is worth to create another MR witch `z_fd_ref/z_fd_unref` return type change to void?

I would suggest we stick we the current behavior.

Reply:
Ok. Cool